crates/uv-fs/src/lib.rs

```diff
@@ -693,6 +693,28 @@ impl LockedFile {
         }
     }
 
+    /// Inner implementation for [`LockedFile::acquire_no_wait`].
+    fn lock_file_no_wait(file: fs_err::File, resource: &str) -> Option<Self> {
+        trace!(
+            "Checking lock for `{resource}` at `{}`",
+            file.path().user_display()
+        );
+        match file.file().try_lock_exclusive() {
```

Comment:
We should loop on `EINTR`, and probably in the blocking cases too, unless I'm missing something that already does that.

Reply:
Deferring to #15996